containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -119,6 +119,28 @@ class UnorderedMapInsertResult {
   uint32_t m_status;
 };
 
+/// \class UnorderedMapInsertOpTypes
+/// \brief Operations applied to the values array upon subsequent insertions.
+/// The default behavior when a k,v pair already exists in the UnorderedMap is
+/// to perform no operation. Alternatively, the caller may select to
+/// instantiate the UnorderedMap with the AtomicAdd insert operator such that
+/// duplicate keys accumulate values into the given values array entry.
+template <class ValueTypeView = int, class ValuesIdxType = uint32_t,
+          class ValueType = int>
+struct UnorderedMapInsertOpTypes {
+  struct NoOp {
+    KOKKOS_FUNCTION
+    void op(ValueTypeView, ValuesIdxType, const ValueType *) const {}
+  };
+  struct AtomicAdd {
+    KOKKOS_FUNCTION
+    void op(ValueTypeView values, ValuesIdxType values_idx,
+            const ValueType *v) const {
+      Kokkos::atomic_add(values.data() + values_idx, *v);
+    }
+  };
+};
```

Comment:
Yes, I agree that we need `ValueTypeView` but I was thinking that you could do 
```suggestion
/// \class UnorderedMapInsertOpTypes
/// \brief Operations applied to the values array upon subsequent insertions.
/// The default behavior when a k,v pair already exists in the UnorderedMap is
/// to perform no operation. Alternatively, the caller may select to
/// instantiate the UnorderedMap with the AtomicAdd insert operator such that
/// duplicate keys accumulate values into the given values array entry.
template <class ValueTypeView = int, class ValuesIdxType = uint32_t>
struct UnorderedMapInsertOpTypes {
  using ValueType = typename ValueTypeView::non_const_value_type;
  struct NoOp {
    KOKKOS_FUNCTION
    void op(ValueTypeView, ValuesIdxType, const ValueType *) const {}
  };
  struct AtomicAdd {
    KOKKOS_FUNCTION
    void op(ValueTypeView values, ValuesIdxType values_idx,
            const ValueType *v) const {
      Kokkos::atomic_add(values.data() + values_idx, *v);
    }
  };
};
```
avoiding `ValueType` to be a template parameter.

Reply:
Ah, certainly.